lib/remap-functions/src/is_blank.rs

```diff
@@ -0,0 +1,156 @@
+use lazy_static::lazy_static;
+use regex::Regex;
+use remap::prelude::*;
+
+lazy_static! {
+    static ref ALL_WHITESPACE_PATTERN: Regex = Regex::new(r"^(\s*)$").unwrap();
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct IsBlank;
+
+impl Function for IsBlank {
+    fn identifier(&self) -> &'static str {
+        "is_blank"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_) | Value::Null),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(IsBlankFn { value }))
+    }
+}
+
+#[derive(Clone, Debug)]
+struct IsBlankFn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for IsBlankFn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        match self.value.execute(state, object)? {
+            Value::Bytes(v) => {
+                let s = &String::from_utf8_lossy(&v)[..];
+
+                if ALL_WHITESPACE_PATTERN.is_match(s) {
+                    Ok(true.into())
+                } else {
+                    match s {
+                        "-" | "\n" => Ok(true.into()),
```

Comment:
Won't the regex catch `\n`?

Reply:
No, it won't. Confirmed by Rustexp.

Reply:
Do we need `\r` in here too then?

The regex docs made me think `\s` would match any whitespace:

> \w, \d and \s are Unicode aware. For example, \s will match all forms of whitespace categorized by Unicode.

https://docs.rs/regex/1.4.2/regex/

Reply:
Okay, I've added carriage return to the pattern